serde/src/main/java/com/amazon/ionhiveserde/caseinsensitivedecorator/IonCaseInsensitiveDecorator.java

```diff
@@ -22,11 +22,11 @@
 public class IonCaseInsensitiveDecorator {
     /**
      * Wraps an IonValue in an IonContainerCaseInsensitiveDecorator if it's an ion container.
-
+     *
      * @return a case insensitive decorator wrapped Ion Value.
      */
     public static IonValue wrapValue(final IonValue v) {
-        if (v == null || v.isNullValue()) {
+        if (v == null) {
```

Comment:
Should behavior the same as the regular Ion container's iterator (Don't need to return null here). Object inspector will validate null values.

Reply:
Thanks for tracking this down :)